src/pages/ResearcherProfile.js

```diff
@@ -350,124 +322,74 @@ export const ResearcherProfile = hh(class ResearcherProfile extends Component {
     });
   };
 
   clearPIData = () => {
     this.setState(prev => {
-      prev.profile.piName = '';
-      prev.profile.piEmail = '';
+      prev.profileProperties.piName = '';
+      prev.profileProperties.piEmail = '';
       return prev;
     });
   };
 
-  saveProfile = (event) => {
-    event.preventDefault();
-    this.setState({ showDialogSave: true });
-  };
-
   cleanObject = (obj) => {
-    // Removes any zero length properties from a copy of the object
-    return omitBy(obj, (s) => { return trim(s.toString()).length === 0; });
-  };
-
-  saveProperties = async (profile) => {
-    await Researcher.createProperties(profile);
-    await this.saveUser();
-    this.setState({ showDialogSubmit: false });
-    this.props.history.push({ pathname: 'dataset_catalog' });
-  };
-
-  dialogHandlerSubmit = (answer) => async () => {
-    if (answer === true) {
-      if (this.state.isResearcher) {
-        let profile = this.state.profile;
-        profile = this.cleanObject(profile);
-        profile.completed = true;
-        if (this.state.profile.completed === undefined) {
-          await this.saveProperties(profile);
-        } else {
-          await this.updateResearcher(profile);
-        }
-      } else {
-        const savedUser = await this.saveUser();
-        this.setState({ isResearcher: savedUser.isResearcher, showDialogSubmit: false });
+    // Removes any empty properties from a copy of the object
+    const propNames = Object.getOwnPropertyNames(obj);
+    for (let propName in propNames) {
+      if (obj[propName] === null || obj[propName] === undefined || obj[propName].toString().length === 0) {
+        delete obj[propName];
       }
-
-    } else {
-      this.setState({ showDialogSubmit: false });
     }
+    return obj;
   };
 
-  updateResearcher = async (profile) => {
-    const profileClone = this.cloneProfile(profile);
-    await Researcher.updateProperties(this.state.currentUser.dacUserId, true, profileClone);
-    await this.saveUser();
-    this.setState({ showDialogSubmit: false });
-    this.props.history.push({ pathname: 'dataset_catalog' });
-  };
-
-  saveUser = async () => {
-    const currentUserUpdate = Storage.getCurrentUser();
-    delete currentUserUpdate.email;
-    currentUserUpdate.displayName = this.state.profile.profileName;
-    currentUserUpdate.additionalEmail = this.state.additionalEmail;
-    currentUserUpdate.roles = this.state.roles;
-    currentUserUpdate.institutionId = this.state.institutionId;
-    const payload = { updatedUser: currentUserUpdate };
-    let updatedUser = await User.update(payload, currentUserUpdate.dacUserId);
-    updatedUser = Object.assign({}, updatedUser, setUserRoleStatuses(updatedUser, Storage));
-    return updatedUser;
-  };
-
-  handleCheckboxChange = (e) => {
-    const value = e.target.checked;
-    this.setState(prev => {
-      prev.profile.checkNotifications = value;
-      return prev;
-    });
+  saveProfile = (event) => {
+    event.preventDefault();
```

Comment:
saveProfile and submit are not new they have just been moved to keep the submit/save handling in one place